src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,49 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+/**
+ * The Duke class runs Duke, the personal assistant chatbot
+ * that helps a person keeps track of various things.
+ */
+
+public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Constructor for class Duke.
+     *
+     * @param filePath File path for saving tasks in the hard disk.
+     * @throws IOException Throws if an unpredicted error occurs.
+     */
+    public Duke(String filePath) throws IOException {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.loadTasks());
+        } catch (DukeException e) {
+            tasks = new TaskList();
+        } catch (FileNotFoundException e) {
+            File file = new File(filePath);
+            file.createNewFile();
```

Comment:
`tasks` is potentially null here. 

Maybe you can use the `finally` block to check if `tasks` is null, and create a new `TaskList` there.